### PR TITLE
Release v2.7.1

### PR DESCRIPTION
**Pull request by antkneed**
Leaving this as a draft until two things are resolved or approved.

1. **blas_impl** does not like any other values aside from `mkl` & `openblas`.

I think there is value in providing an variant that contains the internal GSL CBLAS implementation. The only way I could get the correct **blas_impl** to `conda render ...` was to delete the corresponding section in aggregate's CBC. Even if the local CBC is the same as aggregate's with the additon of `- gslcblas` it would not pick up that option. I resorted to using a new CBC key. 😞 

Any thoughts on what might be is going wrong?

~~2. **osx-arm64**'s `nomkl` build string is fixed in so `openblas` is installable. This should resolve the **osx-arm64** build failure, once the branch used to build 0.3.17 is pushed to openblas-feedstock so we can port [this commit](https://github.com/AnacondaRecipes/openblas-feedstock/commit/accb2dbee2cde139a1c25c45bbbc3b974e764084).~~

```
$ conda search --subdir osx-arm64 --info nomkl
Loading channels: done
nomkl 3.0 hca03da5_0
--------------------
file name   : nomkl-3.0-hca03da5_0.conda
name        : nomkl
version     : 3.0
build       : hca03da5_0                <--- this needs to be 0
build number: 0
size        : 10 KB
license     : BSD
subdir      : osx-arm64
url         : https://repo.anaconda.com/pkgs/main/osx-arm64/nomkl-3.0-hca03da5_0.conda
md5         : e8c8508039dac76724bf1f6e43c77904
timestamp   : 2021-08-09 17:14:34 UTC
dependencies: 
  - blas * openblas
```

(**edit**: `nomkl` for osx-arm64 was updated recently).

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -8,8 +8,8 @@ source:
   url: http://ftp.gnu.org/gnu/gsl/gsl-{{ version }}.tar.gz
   sha256: dcb0fbd43048832b757ff9942691a8dd70026d5da0ff85601e52687f6deeb34b
   patches:
-    # diff from source distribution to https://github.com/ampl/gsl/tree/v2.4.0
-    - patch_cmake_for_windows.diff  # [win]
+    # diff from source distribution to https://github.com/ampl/gsl/tree/v2.7.0
+    - 0001-patch_cmake_for_windows.diff  # [win]
```
Comment (katietz) [inline recipe/meta.yaml]: please add to 'build-requirements' m2-patch dependency for Windows

**Review by katietz — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (antkneed) [reply recipe/meta.yaml]: Thanks! Added this to the build requirements: [see here](https://github.com/AnacondaRecipes/gsl-feedstock/pull/3/commits/6641ec3c31bb203c9a2be3896bf5d98a3133a954)

File: recipe/meta.yaml [hunk 2]
```diff
@@ -24,21 +24,22 @@ requirements:
...
   commands:
     - gsl-config --prefix  # [not win]
     - if not exist %LIBRARY_LIB%\\gsl.lib       exit 1   # [win]
     - if not exist %LIBRARY_LIB%\\gslcblas.lib  exit 1   # [win]
     - if not exist %LIBRARY_BIN%\\gsl.dll       exit 1   # [win]
-    - conda inspect linkages -p $PREFIX $PKG_NAME  # [not win]
-    - conda inspect objects -p $PREFIX $PKG_NAME  # [osx]
 
 about:
   home: http://www.gnu.org/software/gsl/
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: The home url should be with HTTPS

File: recipe/meta.yaml [hunk 3]
```diff
@@ -24,21 +24,22 @@ requirements:
...
-  license: GPL-3.0
+  license: GPL-3.0-or-later
+  license_family: GPL
   license_file: COPYING
   summary: GNU Scientific Library
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:55]: Missing `doc_url: https://www.gnu.org/software/gsl/doc/html/index.html`
Missing `dev_url: https://git.savannah.gnu.org/cgit/gsl.git`

**Review by katietz — APPROVED** [review]
ok.  So, libm should be added for all unix architectures.

**Review by katietz — APPROVED** [review]

**Review by katietz — APPROVED** [review]

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
 
 package:
   name: gsl
   version: {{ version }}
 
 source:
   url: http://ftp.gnu.org/gnu/gsl/gsl-{{ version }}.tar.gz
-  sha256: efbbf3785da0e53038be7907500628b466152dbc3c173a87de1b5eba2e23602b
+  sha256: dcb0fbd43048832b757ff9942691a8dd70026d5da0ff85601e52687f6deeb34b
   patches:
```
Comment (katietz) [inline recipe/meta.yaml]: add here a # [win] too.

**Review by skupr-anaconda — APPROVED** [review]

**Pull request merged**